src/docs/cookbook/maintenance/error-reporting.md

```diff
@@ -47,138 +45,64 @@ To get a DSN, use the following steps:
 
   1. [Create an account with Sentry][].
   2. Log in to the account.
-  3. Create a new app.
-  4. Copy the DSN.
+  3. Create a new Flutter project.
+  4. Copy the code snippet which includes the DSN.
 
 ## 2. Import the Sentry package
 
-Import the [`sentry`][] package into the app.
+Import the [`sentry_flutter`][] package into the app.
 The sentry package makes it easier to send
 error reports to the Sentry error tracking service.
 
 ```yaml
 dependencies:
-  sentry: <latest_version>
+  sentry_flutter: <latest_version>
 ```
 
-## 3. Create a `SentryClient`
+## 3. Initialize the SDK
 
-Create a `SentryClient`. Use the `SentryClient` to send
-error reports to the sentry service.
+Initialize the SDK to capture different unhandled errors automatically.
 
 <!-- skip -->
 ```dart
-final SentryClient _sentry = SentryClient(dsn: "App DSN goes Here");
-```
-
-## 4. Create a function to report errors
-
-With Sentry set up, you can begin to report errors. Since you don't want to
-report errors to Sentry during development, first create a function that
-lets you know whether you're in debug or production mode.
-
-<!-- skip -->
-```dart
-bool get isInDebugMode {
-  // Assume you're in production mode.
-  bool inDebugMode = false;
-
-  // Assert expressions are only evaluated during development. They are ignored
-  // in production. Therefore, this code only sets `inDebugMode` to true
-  // in a development environment.
-  assert(inDebugMode = true);
-
-  return inDebugMode;
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+Future<void> main() async {
+  await SentryFlutter.init(
+    (options) => options.dsn = 'https://example@sentry.io/example',
+    appRunner: () => runApp(MyApp()),
+  );
 }
 ```
 
-Next, use this function in combination with the `SentryClient` to report
-errors when the app is in production mode.
+Alternatively you can pass the DSN to Flutter via Dart define:
 
 <!-- skip -->
-```dart
-Future<void> _reportError(dynamic error, dynamic stackTrace) async {
-  // Print the exception to the console.
-  print('Caught error: $error');
-  if (isInDebugMode) {
-    // Print the full stacktrace in debug mode.
-    print(stackTrace);
-  } else {
-    // Send the Exception and Stacktrace to Sentry in Production mode.
-    _sentry.captureException(
-      exception: error,
-      stackTrace: stackTrace,
-    );
-  }
-}
+```sh
+--dart-define SENTRY_DSN=https://example@sentry.io/example
 ```
 
-## 5. Catch and report Dart errors
+### What does that give me?
 
-Now that you have a function to report errors depending on the environment,
-you need a way to capture Dart errors.
+This is all you need for Sentry to capture unhandled errors in Dart and native layers.  
+Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin and C.
```

Comment:
```suggestion
Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin, C and C++.
```

Reply:
The thing is that it would support Rust, and other native languages, I'm not sure how much is "good enough" to list here.